src/main/webapp/data-visualization.js

```diff
@@ -508,6 +529,142 @@ function toggleDataTable() {
+  bullet.circle.radius = 4;
+
+  const mapTitle = chart.titles.create();
+  let title;
+  const state = stateInfo[location];
+  if (state != null) {
+    title = `${description} in ${state.name} by year`;
+  } else {
+    title = `${description} in ${location} by year`;
+  }
+  mapTitle.text = title;
+
+  chart.legend = new am4charts.Legend();
+  document.getElementById(chartId).style.display = 'block';
+}
+
+// Create yearly state total population growth chart for the submitted query
+async function createYearlyStateTotalChart(
+  personType, action, location, description) {
+  document.getElementById('county-form').style.display = 'block';
+  const loadingMsg = document.getElementById('yearly-total-chart-loading-msg');
+  loadingMsg.innerText = 'Yearly state data chart loading...';
+  const data = [];
+  for (let year = 2010; year <= 2018; year++) {
+    await fetch(getFetchUrl('query', personType, action, location, year) +
+      '&state-total=true')
+      .then((response) => response.json().then((jsonResponse) => ({
+        data: jsonResponse,
+        success: response.ok,
+        status: response.status,
+      })))
+      .then((response) => {
+        if (response.success) {
+          const censusData = removeErroneousData(
+            JSON.parse(response.data.censusData));
+          data.push({
+            'year': year.toString(),
+            'value': parseInt(censusData[1][1]),
+          });
+        }
+      });
+  }
+  displayYearlyChart('yearly-total-chart', data, description, location);
+  loadingMsg.innerText = '';
+}
+
+// Create yearly county population growth chart for the submitted query
+async function createYearlyCountyChart() {
+  const loadingMsg = document.getElementById('yearly-county-chart-loading-msg');
+  loadingMsg.innerText = 'Yearly county data chart loading...';
+  const countyForm = new FormData(document.getElementById('county-form'));
+  const county = countyForm.get('county');
+  const countyDropdown = document.querySelector(
+    '#county option[value=\'' + county + '\']');
+  let countyNumber;
+  if (countyDropdown != null) {
+    countyNumber = countyDropdown.dataset.value;
+  } else {
+    document.getElementById('yearly-county-chart').style.display = 'none';
+    loadingMsg.innerText =
+      'Input must be a county name in the dropdown list';
+    return;
+  }
+  const data = [];
+  const query = getQueryFromHash(window.location.hash);
+  const description = getDescription(query.action, query.personType);
+  const locationDropdown = document.querySelector(
+    '#location option[data-value=\'' + query.location + '\']');
+  let locationInfo;
+  let stateNumber;
+  if (locationDropdown !== null &&
+    !locationDropdown.classList.contains('autocomplete-item')) {
+    // User picked a location from the dropdown
+    stateNumber = query.location;
+  } else { // Have to manually find which state the location is in
+    locationInfo = await findStateOfLocation(query.location);
```

Comment:
this looks similar to some other code in data-viz (which I only remember because I wrote it). Maybe it would make sense to turn it into a function instead of having it written out twice?

Reply:
Done!